app/models/filter.rb

```diff
@@ -180,4 +180,46 @@ def allowed_taxonomies
       errors.add(:location_ids, _('You can\'t assign locations to this resource'))
     end
   end
+
+  def fix_to_current_user_permissions
+    return unless role_id
+    return if permission_ids.empty?
+    user = User.current
+    return unless user
+    return if user.admin?
```

Comment:
how about something like:
```ruby
def need_to_fix_permissions
  role_id.present? && permission_ids.present? && User.current.present? && !User.current.admin?
end
```
and then only [conditionally](http://guides.rubyonrails.org/active_record_callbacks.html#using-if-and-unless-with-a-symbol) run this callback if the condition is true?